integration/auth/auth_test.go

```diff
@@ -260,6 +267,198 @@ func signInWithPassword(email, password string) (string, error) {
+				Rounds:     1,
+				InputOrder: hash.InputOrderSaltFirst,
+			},
+			localHash: sha512.New(),
+		},
+		{
+			testName: "SHA512_PasswordFirst",
+			hashConfig: hash.SHA512{
+				Rounds:     1,
+				InputOrder: hash.InputOrderPasswordFirst,
+			},
+			localHash: sha512.New(),
+		},
+		{
+			testName: "HMAC_MD5_SaltFirst",
+			hashConfig: hash.HMACMD5{
+				Key:        []byte(key),
+				InputOrder: hash.InputOrderSaltFirst,
+			},
+			localHash: hmac.New(md5.New, []byte(key)),
+		},
+		{
+			testName: "HMAC_MD5_PasswordFirst",
+			hashConfig: hash.HMACMD5{
+				Key:        []byte(key),
+				InputOrder: hash.InputOrderPasswordFirst,
+			},
+			localHash: hmac.New(md5.New, []byte(key)),
+		},
+		{
+			testName: "HMAC_SHA1_SaltFirst",
+			hashConfig: hash.HMACSHA1{
+				Key:        []byte(key),
+				InputOrder: hash.InputOrderSaltFirst,
+			},
+			localHash: hmac.New(sha1.New, []byte(key)),
+		},
+		{
+			testName: "HMAC_SHA1_PasswordFirst",
+			hashConfig: hash.HMACSHA1{
+				Key:        []byte(key),
+				InputOrder: hash.InputOrderSaltFirst,
+			},
+			localHash: hmac.New(sha1.New, []byte(key)),
+		},
+		{
+			testName: "HMAC_SHA256_SaltFirst",
+			hashConfig: hash.HMACSHA256{
+				Key:        []byte(key),
+				InputOrder: hash.InputOrderSaltFirst,
+			},
+			localHash: hmac.New(sha256.New, []byte(key)),
+		},
+		{
+			testName: "HMAC_SHA256_PasswordFirst",
+			hashConfig: hash.HMACSHA256{
+				Key:        []byte(key),
+				InputOrder: hash.InputOrderPasswordFirst,
+			},
+			localHash: hmac.New(sha256.New, []byte(key)),
+		},
+		{
+			testName: "HMAC_SHA512_SaltFirst",
+			hashConfig: hash.HMACSHA512{
+				Key:        []byte(key),
+				InputOrder: hash.InputOrderSaltFirst,
+			},
+			localHash: hmac.New(sha512.New, []byte(key)),
+		},
+		{
+			testName: "HMAC_SHA512_PasswordFirst",
+			hashConfig: hash.HMACSHA512{
+				Key:        []byte(key),
+				InputOrder: hash.InputOrderPasswordFirst,
+			},
+			localHash: hmac.New(sha512.New, []byte(key)),
+		},
+	}
+	// h := hash.HMACSHA256{
+	// 	Key: []byte(key),
+	// 	InputOrder: InputOrderSaltFirst
+	// }
+	// myHash := hmac.New(sha256.New, []byte(key))
+	// myHash.Write([]byte(salt+password))
+	for _, test := range tests {
+		hC, _ := test.hashConfig.Config()
+		fmt.Println(hC)
+		if hC["passwordHashOrder"] == "PASSWORD_AND_SALT" {
+			test.localHash.Write([]byte(password + salt))
+		} else {
+			if hC["passwordHashOrder"] == "SALT_AND_PASSWORD" {
+				test.localHash.Write([]byte(salt + password))
+			} else {
+				t.Fatalf("Unexpected value for passwordHashOrder: %s", hC["passwordHashOrder"])
+			}
+		}
+		// TODO: Should this be test.localHash.Sum(make([]byte, 0))  ???
+		user := (&auth.UserToImport{}).
+			UID(uid).
+			Email(email).
+			PasswordHash(test.localHash.Sum(nil)).
+			PasswordSalt([]byte(salt))
+		result, err := client.ImportUsers(context.Background(), []*auth.UserToImport{user}, auth.WithHash(test.hashConfig))
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer deleteUser(uid)
```

Comment:
Have you tried running this test? I'm wondering how it can possibly work. Because `deleteUser()` calls are deferred until all cases are run, it will attempt to import the same `uid` repeatedly, which should fail.

Reply:
Fair point, sorry- this is an artifact from a previous testing structure before I moved to this iterative version.  Fixed.

Reply:
I think this issue still exists. This is using the same `uid` value in multiple iterations of the test. You should either use unique uid+email for each iteration, or not defer the calls to `deleteUser()`.